src/tir/usmp/algo/hill_climb.cc

```diff
@@ -312,6 +331,14 @@ class HillClimbAllocator : public GreedyBase {
     Map<BufferInfo, PoolAllocation> result;
     // return winning combination
     for (auto it : result_pool_allocations) {
+      // post-check that everything was fit
+      const BufferInfoNode* buf = it.first;
+      const PoolAllocation& pa = it.second;
+      if (NullValue<PoolInfo>().same_as(pa->pool_info) ||
+          !IsValidPlacement(pa->pool_info, pa->byte_offset->value, buf->size_bytes->value)) {
+        std::unordered_map<PoolInfo, size_t, ObjectPtrHash, ObjectPtrEqual> m = {};
+        SelectPlacementPool(GetRef<BufferInfo>(buf), m);
```

Comment:
maybe add an error message specifying the required memory size instead of the error message from SelectPlacementPool?

Reply:
The "optimistic" estimation can be calculated from current error message as sum of `size_bytes` (buffer size) and `size_hint_bytes` (pool size)